default.nix

```diff
@@ -1,87 +1,99 @@
-  addWerror = drv: appendConfigureFlag drv "--ghc-option=-Werror";
-  filterSource = drv: drv.overrideAttrs (attrs : {
-    src = builtins.filterSource cleanSourceFilter attrs.src;
-  });
-  cleanSourceFilter = with pkgs.stdenv;
-    name: type: let baseName = baseNameOf (toString name); in ! (
-      # Filter out .git repo
-      (type == "directory" && baseName == ".git") ||
-      lib.hasSuffix "~" baseName ||
-      builtins.match "^\\.sw[a-z]$" baseName != null ||
-      builtins.match "^\\..*\\.sw[a-z]$" baseName != null ||
-      builtins.match "^\\.ghc\\.environment\\..*$" baseName != null ||
-      baseName == "dist" || baseName == "dist-newstyle" ||
-      baseName == "cabal.project.local" ||
-      lib.hasSuffix ".nix" baseName ||
-      lib.hasSuffix ".dhall" baseName ||
-      (type == "symlink" && lib.hasPrefix "result" baseName) ||
-      (type == "directory" && baseName == ".stack-work")
-    );
-  plutusPkgs = (import ./pkgs { inherit pkgs; }).override {
-    overrides = self: super: {
-      # we want to enable benchmarking, which also means we have criterion in the corresponding env
-      # we need to enable benchmarks as a flag too, until we are on a newer nixpkgs where this is implied by doBenchmark
-      language-plutus-core = addWerror (appendConfigureFlag (doBenchmark (doHaddockHydra (addRealTimeTestLogs (filterSource super.language-plutus-core)))) "--enable-benchmarks");
-      plutus-core-interpreter = addWerror (doHaddockHydra (addRealTimeTestLogs (filterSource super.plutus-core-interpreter)));
-      plutus-exe = addWerror (addRealTimeTestLogs (filterSource super.plutus-exe));
-      core-to-plc = addWerror (doHaddockHydra (addRealTimeTestLogs (filterSource super.core-to-plc)));
-      plutus-ir = addWerror (doHaddockHydra (addRealTimeTestLogs (filterSource super.plutus-ir)));
-      plutus-th = addWerror (doHaddockHydra (addRealTimeTestLogs (filterSource super.plutus-th)));
-      plutus-use-cases = addWerror (addRealTimeTestLogs (filterSource super.plutus-use-cases));
-      wallet-api = addWerror (doHaddockHydra (addRealTimeTestLogs (filterSource super.wallet-api)));
+  src = localLib.cleanSourceHaskell ./.;
+  isPlutus = (lib.hasPrefix "plutus") || (lib.hasPrefix "language-plutus");
+  packages = self: ({
+    inherit pkgs;
+
+    # This is the stackage LTS plus overrides, plus the plutus
+    # packages.
+    haskellPackages = self.callPackage localLib.iohkNix.haskellPackages {
+      inherit forceDontCheck enableProfiling enablePhaseMetrics enableHaddockHydra
+        enableBenchmarks fasterBuild enableDebugging enableSplitCheck;
+        pkgsGenerated = ./pkgs;
+        filter = localLib.isPlutus;
+        requiredOverlay = ./nix/overlays/required.nix;
     };
-  };
-  other = rec {
-    tests = let
-      src = localLib.cleanSourceTree ./.;
-    in {
+
+    plutusPackages = localLib.getPackages {
+      inherit (self) haskellPackages; filter = localLib.isPlutus;
```

Comment:
This seems like a weirdly indirect way of getting back our packages. I wouldn't be totally averse to just listing them, tbh.